models/list_type_blackbox_test.go

```diff
@@ -0,0 +1,36 @@
+package models_test
+
+import (
+	"testing"
+
+	. "github.com/almighty/almighty-core/models"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListType_Equal(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+
+	a := ListType{
+		SimpleType:    SimpleType{Kind: KindList},
+		ComponentType: SimpleType{Kind: KindString},
+	}
+
+	// Test type incompatibility
+	assert.False(t, a.Equal(DummyEqualer{}))
+
+	// Test simple type difference
+	b := ListType{
+		SimpleType:    SimpleType{Kind: KindString},
+		ComponentType: SimpleType{Kind: KindString},
+	}
+	assert.False(t, a.Equal(b))
+
+	// Test component type difference
+	c := ListType{
+		SimpleType:    SimpleType{Kind: KindList},
+		ComponentType: SimpleType{Kind: KindInteger},
+	}
+	assert.False(t, a.Equal(c))
```

Comment:
There should be a case where the two objects are equal:
assert.True(t, a.Equal(c))
assert.True(t, c.Equal(a)) // test for inverse